src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1164,11 +1164,15 @@ def process_declarations(self, parent, nodes, arg_list):
 
                 if initialisation:
                     if has_constant_value:
-                        # If it is a parameter parse its initialization
-                        tmp = Assignment()
+                        # If it is a parameter parse its initialization into
+                        # a dummy Assignment inside a Schedule which temporally
+                        # hijacks the parent's node symbol table
+                        tmp_sch = Schedule(symbol_table=parent.symbol_table)
```

Comment:
This is a nice solution but the general approach of creating a dummy parent still feels hacky somehow. This complaint is obviously not for this issue.

Reply:
I agree, process_node() should return the created node instead of having the side-effect of appending it to the given parent, and I suspect thats a difficult change with implications everywhere process_node method is used (the appending will be responsibility of the caller ...).